DEATHS_extract_processing.R

```diff
@@ -0,0 +1,55 @@
+
+#####################################################
+# Draft DEATHS Extract Processing Code
+# Author: Catherine Holland
+# Date: February 2022
+# Written on RStudio Server
+# Version of R - 3.6.1
+# Input -
+# Description - Takes IT extract of Deaths data.
+#               Creates new death date variable,
+#               uses NRS death date as death date if avaliable.
+#               Saves outputted flagged dataset.
+#####################################################
+
+
+## packages ##
+
+library(dplyr)
+library(stringr)
+library(readr)
+
+## Read data in ##
+deaths_file <- read_csv(
+  file = get_it_deaths_path(),
+  col_type = cols(
+    `PATIENT_UPI [C]` = col_character(),
+    `PATIENT DoD DATE (NRS)` = col_date(format = "%d-%m-%Y"),
+    `PATIENT DoD DATE (CHI)` = col_date(format = "%d-%m-%Y")
+  )
+)
+
+names(deaths_file) <- str_replace_all(names(deaths_file), " ", "_")
+
+# rename variables
+deaths_file <- deaths_file %>%
+  rename(
+    chi = "PATIENT_UPI_[C]",
+    death_date_NRS = "PATIENT_DoD_DATE_(NRS)",
+    death_date_CHI = "PATIENT_DoD_DATE_(CHI)"
+  )
+
+## one record per chi ##
+deaths_file <- deaths_file %>%
+  dplyr::distinct(chi, .keep_all = TRUE) %>%
+  # Use the NRS deathdate unless it isn't there
+  mutate(death_date = if_else(is.na(death_date_NRS), death_date_CHI, death_date_NRS))
```

Comment:
`dplyr::coalesce()` does exactly what we want for the last line i.e. pick the first non-missing date.
```suggestion
deaths_file <- deaths_file %>%
  dplyr::distinct(chi, .keep_all = TRUE) %>%
  # Use the NRS deathdate unless it isn't there
  mutate(death_date = dplyr::coalesce(death_date_NRS, death_date_CHI)
```